### ISSUE TITLE
Call API endpoints asynchronously 

### ISSUE THREAD

**Issue by dmitriim** [valentineus/moodle-webhooks#16]
At some points API end points could be slow. This will slow down Moodle operations related to the configured events. 
The solution could be to have a configuration option to run web hooks asynchronously via adhoc tasks or cron task.

**Comment by valentineus** [comment]
Thanks! It's a valuable issue.

I too see this problem. This problem will be to [resolve](https://github.com/valentineus/moodle-webhooks/tree/develop) a new version. I'm using [adhoc_task](https://docs.moodle.org/dev/Task_API#Adhoc_tasks) to asynchrony work.

**Comment by dmitriim** [comment]
Nice! 
Would it be possible to make it configurable and let the admin to decide if a webhook should be processed real time or by adhoc? 
Also would be useful to fall back to adhoc if for some reason the webhook failed during real-time execution.

**Comment by mlohoar** [comment]
I agree that this should definitely be a setting for using adhoc or not.  Otherwise you might get into race conditions between Moodle and the external service.
